mlflow/entities/span.py

```diff
@@ -361,6 +361,9 @@ def to_immutable_span(self) -> "Span":
         return Span(self._span)
 
 
+NO_OP_SPAN_REQUEST_ID = "MLFLOW_NO_OP_SPAN_REQUEST_ID"
```

Comment:
note: This special ID is added for client to distinguish if the API is called with NoOpSpan or not.

For example, let's say we have the following code in model.
```
root_span = client.start_trace(experiment_id=exp_id, name="predict")
child_span = client.start_span(name="child", request_id=root_span.request_id, parent_id=root_span.span_id)
```
When `client.start_trace()` fails with some reason, it returns `NoOpSpan` that represents something went wrong. Previously, NoOpSpan has `None` request ID. Then, the next line `client.start_span()` is called with `None` request ID, and MLflowClient raises invalid parameter exception.

After this change, NoOpSpan returns this special ID, so the client can tell there was some error in upstream, not incorrect user input. So it will just do nothing, instead of raising exception.